pallets/parachain-staking/src/migrations.rs

```diff
@@ -17,19 +17,113 @@
 //! # Migrations
 use crate::{
 	pallet::{migrate_nominator_to_delegator_state, RoundIndex},
-	BalanceOf, CandidateState, CollatorCandidate, CollatorState2, Config, DelegatorState,
-	ExitQueue2, NominatorState2, Points, Round, Staked,
+	BalanceOf, Bond, CandidateState, CollatorCandidate, CollatorState2, Config, DelegatorState,
+	ExitQueue2, NominatorState2, Pallet, Points, Round, Staked,
 };
 #[cfg(feature = "try-runtime")]
 use crate::{Collator2, Delegator, Nominator2};
 #[cfg(feature = "try-runtime")]
+use frame_support::traits::OnRuntimeUpgradeHelpersExt;
+#[cfg(feature = "try-runtime")]
 use frame_support::Twox64Concat;
+extern crate alloc;
+#[cfg(feature = "try-runtime")]
+use alloc::format;
 use frame_support::{
 	pallet_prelude::PhantomData,
 	traits::{Get, OnRuntimeUpgrade},
 	weights::Weight,
 };
-use sp_std::collections::btree_map::BTreeMap;
+use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
+
+/// Migration to properly increase maximum delegations per collator
+/// This migration can be used to recompute the top and bottom delegations whenever
```

Comment:
A couple thoughts on reusing this:

1. The only hard constraint on reusing the same migration is that it have a unique name since this is used as a storage key in `pallet_migrations`.
2. Alternative designs we could consider in the future would be (1) automatically doing this upon the next round change, or (2) having an extrinsic that does this (of course, it will be heavy...)

Reply:
The unique name, of course, would be done in `runtime/common/src/migrations.rs`

Reply:
> automatically doing this upon the next round change

Actually, yeah, this should be done at a round boundary. Otherwise you're going to count a bunch more delegators for the round in which the change happened even though their contributions didn't help the collators get elected in that round.

Reply:
Right, that's "unfair", but isn't significant unless the round is extremely long.